File: Data Structures & Algorithms/median-of-two-sorted-arrays/submission-1.py
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        
        m = len(nums1)
        n = len(nums2)

        na = nums1 + nums2
        na.sort()

        nal = len(na)

        # [1, 2, 3, 4, 5]

        if nal == 0:
            res = 0
        elif nal == 1:
            res = na[0]
        elif nal % 2 == 0:
            res = (na[nal//2 - 1] + na[(nal//2)]) / 2
        elif nal % 2 == 1:
            res = na[(nal-1)//2]
        return res

File: Data Structures & Algorithms/anagram-groups/submission-1.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {

        unordered_map<string, vector<string>> groups;

        vector<vector<string>> res;

        string key;

        for (auto str : strs) {
            key = str;
            sort(key.begin(), key.end());

            groups[key].push_back(str);
        }

        for (auto val : groups) {
            res.push_back(val.second);
        }
        return res;
    }
};
